File: packages/client/ios/RNOta/Watchdog/BootState.swift
import Foundation

public struct BootState: Equatable, Sendable {
  public var schemaVersion: Int
  public var bootAttempt: Int
  public var lastSuccessfulBoot: Int
  public var status: BootStatus
  public var consecutiveFailures: Int
  public var lastFailureReason: BootFailureReason

  public static let empty = BootState(
    schemaVersion: BootPaths.supportedSchemaVersion,
    bootAttempt: 0,
    lastSuccessfulBoot: 0,
    status: .idle,
    consecutiveFailures: 0,
    lastFailureReason: .none
  )

  public init(
    schemaVersion: Int = BootPaths.supportedSchemaVersion,
    bootAttempt: Int = 0,
    lastSuccessfulBoot: Int = 0,
    status: BootStatus = .idle,
    consecutiveFailures: Int = 0,
    lastFailureReason: BootFailureReason = .none
  ) {
    self.schemaVersion = schemaVersion
    self.bootAttempt = bootAttempt
    self.lastSuccessfulBoot = lastSuccessfulBoot
    self.status = status
    self.consecutiveFailures = consecutiveFailures
    self.lastFailureReason = lastFailureReason
  }
}

File: packages/client/android/src/main/java/com/rnota/watchdog/BootState.kt
package com.rnota.watchdog

/**
 * In-memory / on-disk representation of `boot.json` (schema v1).
 *
 * Owned exclusively by [BootWatchdog]. Never written by Bundle Store recovery
 * beyond creating an empty placeholder file.
 */
data class BootState(
  val schemaVersion: Int = BootPaths.SUPPORTED_SCHEMA_VERSION,
  val bootAttempt: Int = 0,
  val lastSuccessfulBoot: Int = 0,
  val status: BootStatus = BootStatus.IDLE,
  val consecutiveFailures: Int = 0,
  val lastFailureReason: BootFailureReason = BootFailureReason.NONE,
) {
  companion object {
    val EMPTY: BootState = BootState()
  }
}
